src/components/contracts/modals/PurchaseModal/PurchaseFormModalPage.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import {
+  TitleWrapper,
+  Title,
+  Form,
+  InputGroup,
+  Row,
+  Input,
+  RightBtn,
+  ContractLink,
+  LeftBtn
+} from '../CreateContractModal.styles';
+
+import {
+  Divider,
+  HeaderFlex,
+  OrderSummary,
+  ProxyRouterContainer,
+  Values,
+  PreviewCont,
+  EditBtn
+} from './common.styles';
+
+import { IconExternalLink } from '@tabler/icons';
+import { formatDuration, formatSpeed, formatPrice } from '../../utils';
+
+export const PurchaseFormModalPage = ({
+  inputs,
+  setInputs,
+  onFinished,
+  contract,
+  rate,
+  pool,
+  explorerUrl,
+  onEditPool,
+  close
+}) => {
+  const [isEditPool, setIsEditPool] = useState(false);
+  const handleInputs = e => {
+    e.preventDefault();
+    setInputs({ ...inputs, [e.target.name]: e.target.value });
+  };
+
+  const handleClose = e => {
+    e.preventDefault();
+    close();
+  };
+
+  return (
+    <>
+      <TitleWrapper>
+        <Title>Purchase Hashpower</Title>
+        <HeaderFlex>
+          <OrderSummary>ORDER SUMMARY</OrderSummary>
+          <ContractLink onClick={() => window.openLink(explorerUrl)}>
+            <span style={{ marginRight: '4px' }}>View contract</span>
+            <IconExternalLink width={'1.4rem'} />
+          </ContractLink>
+        </HeaderFlex>
+        <Divider />
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'space-between',
+            marginTop: '10px'
+          }}
+        >
+          <div>
+            <OrderSummary>Terms</OrderSummary>
+            <Values>
+              {formatSpeed(contract.speed)} TH/s for{' '}
+              {formatDuration(contract.length)}
+            </Values>
+          </div>
+          <div>
+            <OrderSummary>Price</OrderSummary>
+            <Values>
+              {formatPrice(contract.price)} LMR (≈ $
+              {(formatPrice(contract.price) * rate).toFixed(2)} USD)
```

Comment:
do we really need to format price before multiply operation, or maybe after??
```suggestion
              {(formatPrice(contract.price * rate)} USD)
```